controller/engine_controller.go

```diff
@@ -1148,6 +1148,14 @@ func (m *EngineMonitor) acquireRestoringCounter(acquire bool) error {
 }
 
 func (ec *EngineController) syncSnapshotCRs(engine *longhorn.Engine) error {
+	log := ec.logger.WithField("engine", engine.Name)
+
+	// it should have the `volume-head` snapshot at least for synchronizing the snapshot list by the engine monitor
+	// new migration engine might have an empty snapshot list once its state changed to be active and running
+	if len(engine.Status.Snapshots) == 0 {
```

Comment:
No.
When migration finished and the migration engine is running, it could still has an empty snapshot list here if the engine monitor does not finish synchronizing the snapshots.

Reply:
Should we make the migration ready until the snapshot sync instead? probably something needed to update in the monitor? or at least to sync the snapshot list when determining whether the migration is really finished or not?

Reply:
Check if snapshots sync before volume attachment controller set `MigrationNodeID` ""